app/views/layouts/_adv_search_body.html.haml

```diff
@@ -79,7 +79,7 @@
             %label.control-label.col-md-5
               = _("Global search:")
             .col-md-6
-              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:expression][:selected][:typ] == "global"
+              - checked = @edit && @edit[:expression] && @edit[:expression][:selected] && @edit[:search_type] == "global"
```

Comment:
@isimluk @himdel Thank you very much for your comments which are highly appreciated.
Now I'm just not sure what exactly should be done because of more possibilities/assumptions in this conversation:
starting using `@edit[:expression][:selected][:search_type]` ?
or just editing `@edit[:expression][:selected][:typ]` ?
dropping `@edit[:search_type] ` from all of the places of the code?
I will make the change which is needed. I just have to be sure that it will be the best option.

Actually, I don't like the word _"selected"_ in this idea, because selecting a filter to create a new one is not required so this is why I prefer `@edit[:search_type]`. But maybe `@edit[:expression][:selected]...` could work, too, if I made some more changes and if `nil `value (no filter selected) would be for creating non-global filters.